skimage/filters/ridges.py

```diff
@@ -0,0 +1,651 @@
+    if beta1:
+        warn("""Use keyword parameter 'beta' instead of 'beta1' which
+                will be removed in version 0.16.""")
+        beta = beta1
+
+    if beta1:
+        warn("""Use keyword parameter 'gamma' instead of 'beta2' which
+                will be removed in version 0.16.""")
+        gamma = beta2
+
+    # Check (sigma) scales
+    sigmas = np.arange(scale_range[0], scale_range[1], scale_step)
+    if np.any(np.asarray(sigmas) < 0.0):
+        raise ValueError("Sigma values less than zero are not valid")
+
+    # Rescale filter parameters
+    alpha = 2 * alpha ** 2
+    beta = 2 * beta ** 2
+    gamma = 2 * gamma ** 2
+
+    # Get image dimensions
+    ndim = image.ndim
+    shape = image.shape
+
+    # Invert image to detect dark ridges on light background
+    if black_ridges:
+        image = invert(image)
+
+    # Generate empty (n+1)D arrays for storing auxiliary images filtered
+    # at different (sigma) scales
+    filtered_array = np.zeros(sigmas.shape + image.shape)
+    lambdas_array = np.zeros(sigmas.shape + image.shape)
+
+    # Filtering for all (sigma) scales
+    for i, sigma in enumerate(sigmas):
+
+        # Calculate (abs sorted) eigenvalues
+        eigenvalues = compute_hessian_eigenvalues(image, sigma, sorting='abs')
+
+        if ndim == 2:
+
+            # Get Hessian eigenvalues
+            (lambda1, lambda2) = eigenvalues
+
+            # Compute sensitivity to deviation from a blob-like structure
+            r_b = _divide_nonzero(lambda1, lambda2) ** 2
+
+            # Compute sensitivity to areas of high variance/texture/structure
+            r_g = lambda1 ** 2 + lambda2 ** 2
+
+            # Compute output image for given (sigma) scale
+            filtered = (np.exp(-r_b / beta)
+                        * (np.ones(shape) - np.exp(-r_g / gamma)))
+
+            # Store results in (2+1)D matrices
+            filtered_array[i] = filtered
+            lambdas_array[i] = lambda2
+
+        elif ndim == 3:
+
+            # Get Hessian eigenvalues
+            (lambda1, lambda2, lambda3) = eigenvalues
+
+            # Compute sensitivity to deviation from a plate-like structure
+            r_a = _divide_nonzero(lambda2, lambda3) ** 2
+
+            # Compute sensitivity to deviation from a blob-like structure
+            r_b = _divide_nonzero(lambda1,
+                                  np.sqrt(np.abs(lambda2 * lambda3))) ** 2
+
+            # Compute sensitivity to areas of high variance/texture/structure
+            r_g = lambda1 ** 2 + lambda2 ** 2 + lambda3 ** 2
+
+            # Compute output image for given (sigma) scale
+            filtered = ((np.ones(shape) - np.exp(-r_a / alpha))
+                        * np.exp(-r_b / beta)
+                        * (np.ones(shape) - np.exp(-r_g / gamma)))
+
+            # Store results in (n+1)D matrices
+            filtered_array[i] = filtered
+            lambdas_array[i] = np.max([lambda2, lambda3], axis=0)
```

Comment:
Ok this looks harder to unify 2D and 3D but I still think it can be done. =D Let me think about it...

Reply:
Ok here it is, I think this is right:

```python
       lambda1, *lambdas = compute_hessian_eigenvalues(image, sigma, sorting='abs')
       r_a = np.inf if ndim == 2 else _divide_nonzero(*lambdas) ** 2
       filtered_raw = np.abs(np.multiply.reduce(lambdas)) ** (1/len(lambdas))
       r_b = _divide_nonzero(lambda1, filtered_raw) ** 2
       r_g = sum([lambda1 ** 2] + [lambdai ** 2 for lambdai in lambdas])
       filtered = ((1 - np.exp(-r_a / alpha_sq)) * np.exp(-r_b / beta_sq)
                   * (1 - np.exp(-r_g / gamma_sq)))
       filtered_array[i] = filtered
       lambdas_array[i] = np.max(lambdas, axis=0)
```

Reply:
Very nice. I checked it and its equivalent to the separate 2D/3D cases, so I incorporated it into the PR. Thanks!